src/definitions/modules/video.js

```diff
@@ -83,196 +104,448 @@ $.fn.video = function(parameters) {
+              $muteButton.addClass(settings.className.active);
+            } 
+            else {
+              $volumeChangeButton.removeClass(settings.className.disabled);
+              $volumeProgress.removeClass(settings.className.disabled);
+              $muteButton.removeClass(settings.className.active);
+            }
+            $volumeProgress.progress({ percent: volume * 100 });
+          },
+          rate: function() {
+            var rate = module.get.playbackRate();
+            module.debug('Update playBack rate', rate);
+            $rateInput.val(rate);
+          },
+          readyState: function() {
+            var state = module.get.readyState();
+            module.debug('Update ready state', state);
+            $readyStateRadio.filter('[value=' + state + ']').prop('checked', true);
+          },
+          networkState: function() {
+            var state = module.get.networkState();
+            module.debug('Update network state', state);
+            $networkStateRadio.filter('[value=' + state + ']').prop('checked', true);
+          },
+          timeLookup: function(event) {
+            // virtual time indicator based on the $timeRange "dragged" time
+            var time = module.get.timeRangeValue();
+            module.debug('Update timelookup state', time);
+            $timeLookupBuffer.prop('checked', module.is.timeBuffered(time));
+            $timeLookupPlayed.prop('checked', module.is.timePlayed(time));
+            $timeLookupValue.text(module.get.readableTime(time));
           }
         },
-
-        generate: {
-          // generates iframe html
-          html: function(source, id, url) {
-            module.debug('Generating embed html');
-            var
-              html
-            ;
-            // allow override of settings
-            source = source || settings.source;
-            id     = id     || settings.id;
-            if((source && id) || url) {
-              if(!source || !id) {
-                source = module.get.source(url);
-                id     = module.get.id(url);
-              }
-              if(source == 'vimeo') {
-                html = ''
-                  + '<iframe src="//player.vimeo.com/video/' + id + '?=' + module.generate.url(source) + '"'
-                  + ' width="100%" height="100%"'
-                  + ' frameborder="0" webkitAllowFullScreen mozallowfullscreen allowFullScreen></iframe>'
-                ;
-              }
-              else if(source == 'youtube') {
-                html = ''
-                  + '<iframe src="//www.youtube.com/embed/' + id + '?=' + module.generate.url(source) + '"'
-                  + ' width="100%" height="100%"'
-                  + ' frameborder="0" webkitAllowFullScreen mozallowfullscreen allowFullScreen></iframe>'
-                ;
-              }
+        
+        // the functions in this range will change the video element state, with no relevant return value
```

Comment:
or should they return a true/false bit on the success of the operation ? It would be implemented through try/catch blocks for each dom operation I guess.